src/modules/hip/kernel/non_silent_region_detection.hpp

```diff
@@ -0,0 +1,401 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+// -------------------- Set 0 -  moving mean square kernel device helpers --------------------
+
+// calculate the position in shared memory to avoid bank conflicts
+__host__ __device__ int smem_pos(int pos)
+{
+    return pos + (pos >> 5); // since shared memory banks considered is 32
+}
+
+__device__ float square(float value)
+{
+    return (value * value);
+}
+
+__device__ void compute_prefix_sum(float *input, uint bufferLength)
+{
+    int offset = 1;
+    int tid = hipThreadIdx_x;
+
+    for (int d = bufferLength >> 1; d > 0; d >>= 1)
+    {
+        __syncthreads();
+        for (int idx = tid; idx < d; idx += hipBlockDim_x)
+        {
+            int ai = offset * (2 * idx + 1) - 1;
+            int bi = offset * (2 * idx + 2) - 1;
+            input[smem_pos(bi)] += input[smem_pos(ai)];
```

Comment:
Make this loop one line by changing loop params and condition to process as twice idx. 
idx += twiceBlockDimX (where twiceBlockDimX = 2 * hipBlockDim_x is precomputed)

```
for()
    input[smem_pos(offset * twiceIdx + <A>)] += input[smem_pos(offset * twiceIdx + <B>)];
```
In the above, A and B would ideally be offset -1 and 2*offset -1 and could be outside the inner loop.
That (offset * twiceIdx) could also be made part of the loop condition so that it becomes one variable.